### ISSUE TITLE
Error in feature_importance

### ISSUE THREAD

**Issue by kadyb** [ModelOriented/DALEX#181]
I encountered an error in the feature_importance function. The binary classification model was created in Keras.

```r
library("keras")
library("DALEX")

values1 = runif(10)
values2 = runif(10)
class = as.factor(rep(c("A", "B"), each = 5))
array_2d = array(c(class, values1, values2), dim = c(10, 3))
labels = to_categorical(as.integer(class) - 1)

build_model = function() {
  
  model = keras_model_sequential() 
  
  model %>%
    layer_dense(units = 2, activation = 'sigmoid', input_shape = 2)
  
  model %>%
    compile(
      optimizer = "adam",
      loss = "binary_crossentropy",
      metrics = list("accuracy")
    )
}

array_2d = array_2d[, -1] # drop factor column
model = build_model()

model %>% fit(
  x = array_2d,
  y = labels,
  verbose = 0
)

explainer = explain(model = model, data = as.matrix(array_2d), y = as.integer(class),
                    type = "classification")
vi_explainer = feature_importance(explainer)
```

Output from explain function:
> Preparation of a new explainer is initiated
>   -> model label       :  python.builtin.object  (  default  )
>   -> data              :  10  rows  2  cols 
>   -> target variable   :  10  values 
>   -> model_info        :  package Model of class: keras.engine.sequential.Sequential package unrecognized , ver. Unknown , task regression (  default  ) 
>   -> model_info        :  type set to  classification 
>   -> predict function  : yhat.default will be used (  default  )
>   -> predicted values  :  numerical, min =  0.4553299 , mean =  0.4691292 , max =  0.4925557  
>   -> residual function :  difference between y and yhat (  default  )
>   -> residuals         :  numerical, min =  0.5142212 , mean =  1.030871 , max =  1.54467  
>   A new explainer has been created! 

Error from feature_importance function:
> Error in rbind(deparse.level, ...) : 
>   numbers of columns of arguments do not match
> In addition: Warning messages:
> 1: In mean.default(newX[, i], ...) :
>   argument is not numeric or logical: returning NA
> 2: In mean.default(newX[, i], ...) :
>   argument is not numeric or logical: returning NA
> 3: In data.frame(variable = rep(rownames(raw), ncol(raw)), permutation = rep(seq_len(B),  :
>   row names were found from a short variable and have been discarded

**Comment by hbaniecki** [comment]
Hi, 
I added names for the variables.
```r
library("keras")
library("DALEX")

values1 = runif(10)
values2 = runif(10)
class = as.factor(rep(c("A", "B"), each = 5))
array_2d = array(c(class, values1, values2), dim = c(10, 3))
labels = to_categorical(as.integer(class) - 1)

build_model = function() {
  
  model = keras_model_sequential() 
  
  model %>%
    layer_dense(units = 2, activation = 'sigmoid', input_shape = 2)
  
  model %>%
    compile(
      optimizer = "adam",
      loss = "binary_crossentropy",
      metrics = list("accuracy")
    )
}

colnames(array_2d) <- c("y", "x1", "x2")

array_2d = array_2d[, -1] # drop factor column
model = build_model()

model %>% fit(
  x = array_2d,
  y = labels,
  verbose = 0
)

explainer = explain(model = model, data = as.matrix(array_2d), y = as.integer(class),
                    type = "classification")
vi_explainer = feature_importance(explainer)

plot(vi_explainer, show_boxplots = FALSE)
```
I guess that we could add another check to `explain()`.

**Comment by kadyb** [comment]
Adding names to variables solves this problem. Thanks.

**Comment by pbiecek** [comment]
thanks!
fixed in f835400
